src/Land/Model/soil_heat.jl

```diff
@@ -194,22 +194,15 @@ function compute_gradient_flux!(
         thermal_conductivity(κ_dry, kersten, κ_sat) * ∇transform.soil.heat.T
 end
 
-function flux_second_order!(
-    land::LandModel,
-    soil::SoilModel,
-    heat::SoilHeatModel,
-    flux::Grad,
-    state::Vars,
-    diffusive::Vars,
-    hyperdiffusive::Vars,
-    aux::Vars,
-    t::Real,
-)
+struct DiffHeatFlux <: TendencyDef{Flux{SecondOrder}} end
+struct DiffWaterFlux <: TendencyDef{Flux{SecondOrder}} end
+
+function flux(::InternalEnergy, ::DiffHeatFlux, ::LandModel, args)
+    return -args.diffusive.soil.heat.κ∇T
+end
+
+function flux(::InternalEnergy, ::DiffWaterFlux, ::LandModel, args)
```

Comment:
I think the arguments here should be:  (prog::InternalEnergy, tend::DiffusiveWaterFlux, land::LandModel, args)

how does the line immediately below this (205) know that land is the third argument?

Reply:
Yeah, we need `land::LandModel`. It looks like `tend` is not used, and `prog` is generally not used. I'll update to include the `land` argument.

Reply:
Fixed... Hopefully that makes more sense